feat: compose installation onboarding

cdrage opened the pull request:
feat: compose installation onboarding

### What does this PR do?

* Adds compose installation onboarding, being able to check if you have
  compose installed, install compose as well as continue to the settings
  page.
* Ability to pick and choose your compose installation version from a
  quickpick prompt

### Screenshot/screencast of this PR

<!-- Please include a screenshot or a screencast
explaining what is doing this PR -->



https://github.com/containers/podman-desktop/assets/6422176/035e3b27-6a8b-4dfc-9e73-c1aa0dd98349





### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop/issues/2411
Closes https://github.com/containers/podman-desktop/issues/4181
Closes https://github.com/containers/podman-desktop/issues/4180

### How to test this PR?

<!-- Please explain steps to reproduce -->

Two scenarios:

Installed but not systemwide:
1. Remove docker-compose system-wide (whereis docker-compose)
2. Go through onboarding process, it will skip the installation section
   and just show you the system-wide install button

Not installed:
1. Remove docker-compose from the
   ~/.local/share/containers/podman-desktop/extensions-storage/compose
   folder as well as system-wide
2. Go through the onboarding process

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

cdrage commented:
Must be merged before this PR:
- [ ] https://github.com/containers/podman-desktop/issues/3810
- [ ] https://github.com/containers/podman-desktop/issues/4141

TODO:
- [ ] Write tests.


Otherwise, ready for an initial review / see if the implementation is correct!

I have opened up the other following issues that will come in later PR's:

- https://github.com/containers/podman-desktop/issues/4180
- https://github.com/containers/podman-desktop/issues/4181

cdrage commented:
FYI! We still have steps referencing the "status bar" for compose which will be removed in the future https://github.com/containers/podman-desktop/issues/3809

cdrage commented:
Ready for review @dgolovin @lstocchi @benoitf 

I've added tests, updated the main description with an updated recording as well.

There is a minor bug with re-rendering the version when selecting a new version, but otherwise this is pretty close to what was mocked up in Penpot. There are minor UI styling issues vs what was mocked up but we'll most likely close the UI issue at a later date when it's 100% finalized / implemented.

cdrage commented:
SonarCloud test I'm unable to mark as a false-positive, the failure / warning is regarding the makeExecutable code which had been copied over from the compatibility code.

Also, there is only a few lines of code copied over from the "compose status bar extension" files. However, that code will be removed when we close #3809

lstocchi reviewed (commented):
My feedback after a first play.
The installation of compose works fine. i still have the problem about the system-wide installation but i see https://github.com/containers/podman-desktop/pull/4175 is still open. So after merging that it should work.

The thing i don't like 

1) the kind of content - Do we need to explain how to use compose? Cannot we just add a link so if the user is interested can go there?
2) It's not very clear what should happen when i select the compose version in the first page. 
By clicking on "Want to install a different version?" I expect to select the version and install it immediately but after selecting it and clicking on `next` i see "compose is installing". This puzzled me. Could we make the message a bit more clear in the first page?` Compose will be installed in the next step. Which version do you want to install? (default 2.22.0) ` (or something like that, i'm not the best person to suggest how to write in english lol)
3) if i start the onboarding having compose installed (not system-wide) i've been asked to repeat the workflow from the first step. Could i just be moved to the last "system-wide enablement" step? 

Then i see a lot of UI defects (some of them related to all onboarding and i can help fixing them):
1) full width - i think we should add a max-width
2) the font - mainly for the code tag, we should handle it differently

The other thing that it's weird is when you are at the last page, the title says "compose successfully installed" but then the onboarding workflow is not considered completed because compose has not been set up system-wide.

cdrage commented:
> My feedback after a first play. The installation of compose works fine. i still have the problem about the system-wide installation but i see #4175 is still open. So after merging that it should work.
> 
> The thing i don't like
> 
> 1. the kind of content - Do we need to explain how to use compose? Cannot we just add a link so if the user is interested can go there?

Followed the UI mockup. We are making lots of changes to the onboarding / compose documentation at the moment, so I don't think it's a good idea to link to the website just yet.

> 2. It's not very clear what should happen when i select the compose version in the first page.
>    By clicking on "Want to install a different version?" I expect to select the version and install it immediately but after selecting it and clicking on `next` i see "compose is installing". This puzzled me. Could we make the message a bit more clear in the first page?`Compose will be installed in the next step. Which version do you want to install? (default 2.22.0)` (or something like that, i'm not the best person to suggest how to write in english lol)

I agree! I've updated the PR to reflect that so it's easier to read / understand what's happening how clicking next will install it.

> 3. if i start the onboarding having compose installed (not system-wide) i've been asked to repeat the workflow from the first step. Could i just be moved to the last "system-wide enablement" step?

Looks like a bug with detectComposeInstallation, I've updated that in the commit.

> 
> Then i see a lot of UI defects (some of them related to all onboarding and i can help fixing them):
> 
> 1. full width - i think we should add a max-width
> 2. the font - mainly for the code tag, we should handle it differently



> 
> The other thing that it's weird is when you are at the last page, the title says "compose successfully installed" but then the onboarding workflow is not considered completed because compose has not been set up system-wide.

Updated the last page to reflect that you should install it system wide.


I've opened up other issues to keep track of what we need to update:

- https://github.com/containers/podman-desktop/issues/4321
- https://github.com/containers/podman-desktop/issues/4322
- https://github.com/containers/podman-desktop/issues/4323

lstocchi reviewed (approved):
Thanks for opening the other UI issues. I think those are the only missing pieces to have this working fine.
 
The first page is still too overwhelming for me but it's just a personal taste, i saw that the mockup is like that. Maybe we can refine it in followup PRs if someone else agree.

The rest works fine. I can install different compose versions and everything works on windows.
LGTM!! great job 🚀

dgolovin reviewed (approved):
It is a good first iteration worth to release.
